openstack/novarc

```diff
@@ -25,6 +25,21 @@ if [ -n "`juju config keystone ssl_cert`" ]; then
 elif ((`jq -r '.applications[]| select(."charm-name"=="vault")' $juju_status_json_cache| wc -l`)); then
     # Vault-based ssl
     if `jq -r '.applications.vault.relations.certificates[]' $juju_status_json_cache| grep -q keystone`; then
+        readarray -t VAULT_STATUS < <(juju status vault --format=json | jq -r '.applications.vault.units."vault/0"."workload-status" | .current,.message')
+        if [[ "${VAULT_STATUS[0]}" == "blocked" && "${VAULT_STATUS[1]}" == "Vault needs to be initialized" ]]; then
+            read -p "$(cat <<-END
+This deployment is using vault-based ssl certificates"
+but the vault seems to be not initialized yet.
+It is recommended to initialize it via running
+./tools/vault-unseal-and-authorise.sh before doing anything else.
+Do you want to proceed anyway? (y/n): 
+END
+)" yn
+            case $yn in
+                [nN] ) echo "exiting...";
+                    exit;;
```

Comment:
Since you typically source this file an `exit` will terminate your shell. Using `return` behaves a bit nicer:

```suggestion
                    return;;
```

Reply:
done